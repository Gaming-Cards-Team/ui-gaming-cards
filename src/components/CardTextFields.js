import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Card from "react-bootstrap/Card";
import {RARITY, CARD,CARD_TYPE,AFFILIATION,COLOR,POINT_VALUES,UNIQUENESS,DESCRIPTION} from "../util/cardTextFieldsConstant";

const cardStyle = {
  paddingTop: 10,
  paddingBottom: 15,
};

const attributesStyle = {
  borderRight: "1px solid #a09797e6"
};
const titleStyle = {
  color: "#2d963f",
  fontSize: "xx-large",
  fontWeight: "bold",
};
const subtitleStyle = {
  fontWeight: "bold",
}

const ProductText = ({rarity, position, type, affiliation, color, points, uniqueness, text}) => (
  <div>
    <span style={titleStyle}> Destroy the Death Start</span>
    <h4>Covert Missions</h4>
    <div style={cardStyle}>
      <Card>
        <Card.Body>
          <Row>
            <Col xs={6} style={attributesStyle}>
              <p><span style={subtitleStyle}>{RARITY}</span>{rarity}</p>
              <p><span style={subtitleStyle}>{CARD}</span>{position}</p>
              <p><span style={subtitleStyle}>{CARD_TYPE}</span>{type}</p>
              <p><span style={subtitleStyle}>{AFFILIATION}</span>{affiliation}</p>
              <p><span style={subtitleStyle}>{COLOR}</span>{color}</p>
              <p><span style={subtitleStyle}>{POINT_VALUES}</span> {points}</p>
              <p><span style={subtitleStyle}>{UNIQUENESS}</span>{uniqueness}</p>
            </Col>
            <Col xs={6}>
            <p style={subtitleStyle}>{DESCRIPTION}</p>
              {text}
            </Col>
          </Row>
        </Card.Body>
      </Card>
    </div>
  </div>
);

export default ProductText;
