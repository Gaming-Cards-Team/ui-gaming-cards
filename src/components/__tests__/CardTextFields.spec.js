import React from 'react';
import CardTextFields from '../CardTextFields';
import renderer from 'react-test-renderer';

describe('Test Product Text component', () => {
    it('should validate the component', () => {
        const productDetail = renderer.create(<CardTextFields />).toJSON();

        expect(productDetail).toMatchSnapshot();
    });
});