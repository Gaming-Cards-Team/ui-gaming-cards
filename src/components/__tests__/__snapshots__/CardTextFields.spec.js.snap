// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product Text component should validate the component 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#2d963f",
        "fontSize": "xx-large",
        "fontWeight": "bold",
      }
    }
  >
     Destroy the Death Start
  </span>
  <h4>
    Covert Missions
  </h4>
  <div
    style={
      Object {
        "paddingBottom": 15,
        "paddingTop": 10,
      }
    }
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-6"
            style={
              Object {
                "borderRight": "1px solid #a09797e6",
              }
            }
          >
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Rarity: 
              </span>
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Card#: 
              </span>
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Card type: 
              </span>
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Affiliation: 
              </span>
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Color: 
              </span>
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Point value(s): 
              </span>
               
            </p>
            <p>
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Uniqueness: 
              </span>
            </p>
          </div>
          <div
            className="col-6"
          >
            <p
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Description: 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
